clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -469,13 +505,13 @@ public void testFilterToBatchDiscard() {
         buffer.flip();
 
         ByteBuffer filtered = ByteBuffer.allocate(2048);
-        MemoryRecords.readableRecords(buffer).filterTo(new TopicPartition("foo", 0), new MemoryRecords.RecordFilter() {
+        MemoryRecords.readableRecords(buffer).filterTo(new TopicPartition("foo", 0), new MemoryRecords.RecordFilter(0, 0) {
             @Override
-            protected BatchRetention checkBatchRetention(RecordBatch batch) {
+            protected BatchRetentionResult checkBatchRetention(RecordBatch batch) {
                 // discard the second and fourth batches
                 if (batch.lastOffset() == 2L || batch.lastOffset() == 6L)
-                    return BatchRetention.DELETE;
-                return BatchRetention.DELETE_EMPTY;
+                    return new BatchRetentionResult(BatchRetention.DELETE, false);
+                return new BatchRetentionResult(BatchRetention.DELETE_EMPTY, true);
```

Comment:
Since there is no marker, It seems that containsMarkerForEmptyTxn should be false.